gradio/components/file_explorer.py

```diff
@@ -0,0 +1,215 @@
+"""gr.FileExplorer() component"""
+
+from __future__ import annotations
+
+import itertools
+import os
+import re
+from glob import glob as glob_func
+from pathlib import Path
+from typing import Callable, Literal
+
+from gradio_client.documentation import document, set_documentation_group
+from gradio_client.serializing import JSONSerializable
+
+from gradio.components.base import IOComponent, server
+from gradio.events import (
+    Changeable,
+    EventListenerMethod,
+)
+
+set_documentation_group("component")
+
+
+@document()
+class FileExplorer(Changeable, IOComponent, JSONSerializable):
+    """
+    Creates a file component that allows uploading generic file (when used as an input) and or displaying generic files (output).
+    Preprocessing: passes the uploaded file as a {tempfile._TemporaryFileWrapper} or {List[tempfile._TemporaryFileWrapper]} depending on `file_count` (or a {bytes}/{List{bytes}} depending on `type`)
```

Comment:
```suggestion
    Preprocessing: passes the selected file as a {str} absolute path or {list[str}} depending on `file_count`
```

Reply:
This can actually be either a file or directory and the path is relative to root. Although it should (and might already) accept an absolute path as well. I haven't tested that.

I changed to 

```
    Preprocessing: passes the selected file or directory as a {str} path (relative to root) or {list[str}} depending on `file_count`
```

Reply:
Supporting `.` will probably need some frontend changes but very simple to implement.